include/seqan3/alignment/pairwise/detail/policy_optimum_tracker.hpp

```diff
@@ -66,6 +66,70 @@ struct max_score_updater
     }
 };
 
+/*!\brief A function object that compares and possibly updates the alignment optimum with the current cell.
+ * \ingroup pairwise_alignment
+ *
+ * \details
+ *
+ * Updates the current alignment optimum with the new score and the respective coordinate if the new score
+* is greater or equal to the score of the current optimum and the cell is either in the target row or column.
+ * For the banded alignment this additional check is needed because the band might not reach the last row of the matrix for all columns. 
+ * Thus, the last cell of a column will only be checked if the corresponding coordinate points to the last row of the matrix.
```

Comment:
Sorry I don't get it.. So from the code I understand that the score is only updated if the input cell *is in the target_row / target_column*. But the documentation here and the naming `max_score_banded_updater` suggests that this function is *in general* updating the score. But what about the inner matrix cells? They will not be updated here. Are there multiple "score_updaters"?

Reply:
Yes, there can be different updater functions. These are basically binary functions that compare and update the current score. So there is also another one that is used by regular global and the local alignment and the correct update function is set by the alignment configuration. I will also try to make a little picture to make it more clear what is meant.

Reply:
And later for the local alignment we can use the regular max updater because we look at every cell anyway.